l10n_it_sale_commission_riba/wizard/wizard_riba_commissions_settle.py

```diff
@@ -0,0 +1,107 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import date, timedelta
+from odoo import models, api, _
+
+
+class SaleCommissionMakeSettle(models.TransientModel):
+    _inherit = "sale.commission.make.settle"
+
+    @api.multi
+    def action_settle(self):
+        """
+        Esclude dal calcolo delle commissioni le fatture insolute, quelle con termini
+        di pagamento Ri.Ba salvo buon fine se non sono passati almeno 5 giorni dalla
+        data di scadenza e quelle che hanno impostato manualmente il flag
+        'no_commission'(per esempio se ormai è insoluta da anni).
+        Il metodo predefinito non prevede hooks quindi è necessario riscriverlo
```

Comment:
Rompere la catena di ereditarietà è considerato un _antipattern_ che non mi sento di approvare :exclamation: 
Piuttosto propenderei nel proporre una modifica al modulo https://github.com/OCA/commission/tree/12.0/sale_commission per aggiungerlo. Se ho ben capito, dovrebbe bastarti un metodo `get_agent_lines(agent, date_to_agent) `.  

Altra nota: i commenti è preferibile scriverli in inglese.

Grazie!

Reply:
Hai ragione riguardo all'ereditarietà, l'avevo scritto anche nei commenti che eravamo stati costretti.
Effettivamente basterebbe recuperare il dominio della ricerca alla riga 81 con un metodo, che lascerebbe la possibilità di ridefinirlo da un altro modulo: https://github.com/OCA/commission/blob/12.0/sale_commission/wizard/wizard_settle.py#L81

Suggerisci di aprire una PR o una issue nel repo `commission`?
La versione è datata non so se le considerino...

> Altra nota: i commenti è preferibile scriverli in inglese.
avevamo visto che altri moduli di `l10n-italy` hanno commenti in italiano e siccome anche questo modulo può servire solo in Italia abbiamo preferito metterli in italiano per semplicità, però in linea generale hai ragione e se dovessimo pubblicare dei moduli un po' più generici di sicuro li metteremmo in inglese

Reply:
> Suggerisci di aprire una PR o una issue nel repo `commission`?

una PR, poi aggiungi il link in questo thread. Grazie!

Reply:
Ciao Alex,
abbiamo aperto la PR in `commission`
https://github.com/OCA/commission/pull/406